File: RaspberryServer2/src/AppClasses/ActionClass.cpp
#include "ActionClass.h"

ActionClass::ActionClass(){
    //ctor
}

ActionClass::~ActionClass(){
    //dtor
}



string ActionClass::parseToJson(){
    DynamicJsonBuffer jsonBuffer;

    JsonArray& root = jsonBuffer.createArray();
    for (unsigned short i = 0; i < this->actions.size(); i++){
//        cout << "\nParse: " << this->actions.at(i).parseToJson();
        root.add(this->actions.at(i).parseToJson());
    }

    string json;
    root.printTo(json);
    return json;

}

bool ActionClass::createFromJson(string json){
    DynamicJsonBuffer jsonBuffer;
    JsonArray& root = jsonBuffer.parseArray(json);
    for (unsigned short i = 0; i < root.size(); i++) {
        TaskButtonClass taskButton = TaskButtonClass();
        taskButton.createFromJson(root[i]);
        this->actions.insert(this->actions.end(), taskButton);
    }

    return true;

}

vector<TaskButtonClass> ActionClass::getActions(){
    return this->actions;
}
void ActionClass::setActions(vector<TaskButtonClass> _actions){
    this->actions = _actions;
}

File: AndroidClient/app/src/main/java/androidclient/automacaoz/raspberry/bruno/azandroidclient/AppClasses/ActionClass.java
package androidclient.automacaoz.raspberry.bruno.azandroidclient.AppClasses;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Bruno on 18/06/2017.
 */

public class ActionClass {
    private final String TAG = "ACTION_CLASS";
    List<TaskButtonClass> actions;

    public ActionClass(List<TaskButtonClass> actions) {
        this.actions = actions;
    }
    public ActionClass(String jsonArray){
        JSONArray array = null;
        try {
            array = new JSONArray(jsonArray);
        } catch (JSONException e) {
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return;
        }

        actions = new ArrayList<>();
        try {
            for (int i = 0; i < array.length(); i++) {
                TaskButtonClass taskButton = new TaskButtonClass(array.get(i).toString());
                actions.add(taskButton);
            }
        } catch (JSONException e) {
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return;
        }


    }

    public List<TaskButtonClass> getActions() {return actions;}
    public void setActions(List<TaskButtonClass> actions) {this.actions = actions;}

    public String parseToJson(){
        JSONArray jsonArray = new JSONArray();

        for (int i = 0; i < actions.size(); i++){
            try {
                jsonArray.put(new JSONObject(actions.get(i).parseToJson()));
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }

        return jsonArray.toString();
    }
}
